keras/layers/core.py

```diff
@@ -566,6 +566,11 @@ def get_output_shape_for(self, input_shape):
                 else:
                     return K.int_shape(x)
             # otherwise, we default to the input shape
+            warnings.warn("`output_shape` argument not specified for layer {} "
+                          "and cannot be automatically inferred with the Theano backend. "
+                          "Defaulting to output shape `{}` (same as input shape). "
+                          "If the expected output shape is different, specify it via the `output_shape` argument."
+                          .format(self.name, input_shape))
```

Comment:
For consistency with the rest of the file, use `'` instead of `"` to format strings.